PHPStan false positives with phpstan version 2.1.45

<!--
⚠️ Public content warning ⚠️
Issues created in this repository will be publicly available on the Internet. Please do not include any sensitive or internal information in your issue.

🔒 Reporting security vulnerabilities 🔒
Do not create issues about security vulnerabilities. Please consult our security policy for instructions about how to safely report a security issue: https://github.com/az-digital/az_quickstart/security/policy
-->

## Problem/Motivation
New builds of az_quickstart are having PHPStan failures since the release of [phpstan 2.1.45](https://github.com/phpstan/phpstan/releases/tag/2.1.45) yesterday.

### Describe the bug

```
 ------ ------------------------------------------------------------------------------------------------------------------------------- 
  Line   web/profiles/custom/az_quickstart/modules/custom/az_accordion/src/Plugin/Field/FieldFormatter/AZAccordionDefaultFormatter.php  
 ------ ------------------------------------------------------------------------------------------------------------------------------- 
  98     PHPDoc tag @var with type Drupal\az_accordion\Plugin\Field\FieldType\AZAccordionItem is not subtype of native type             
         Drupal\Core\Field\FieldItemInterface.                                                                                          
         🪪  varTag.nativeType                                                                                                          
 ------ ------------------------------------------------------------------------------------------------------------------------------- 

 ------ --------------------------------------------------------------------------------------------------------------------- 
  Line   web/profiles/custom/az_quickstart/modules/custom/az_card/src/Plugin/Field/FieldFormatter/AZCardDefaultFormatter.php  
 ------ --------------------------------------------------------------------------------------------------------------------- 
  110    PHPDoc tag @var with type Drupal\az_card\Plugin\Field\FieldType\AZCardItem is not subtype of native type             
         Drupal\Core\Field\FieldItemInterface.                                                                                
         🪪  varTag.nativeType                                                                                                
 ------ --------------------------------------------------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------------------------------------------------- 
  Line   web/profiles/custom/az_quickstart/modules/custom/az_publication/src/Plugin/Field/FieldWidget/AZEntityRoleInlineFormComplex.php  
 ------ -------------------------------------------------------------------------------------------------------------------------------- 
  97     Access to an undefined property Drupal\Core\Field\FieldItemInterface::$entity.                                                  
         🪪  property.notFound                                                                                                           
         💡  Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property                                           
 ------ -------------------------------------------------------------------------------------------------------------------------------- 

 ------ --------------------------------------------------------------------------------------------------------------------------- 
  Line   web/profiles/custom/az_quickstart/modules/custom/az_ranking/src/Plugin/Field/FieldFormatter/AZRankingDefaultFormatter.php  
 ------ --------------------------------------------------------------------------------------------------------------------------- 
  110    PHPDoc tag @var with type Drupal\az_ranking\Plugin\Field\FieldType\AZRankingItem is not subtype of native type             
         Drupal\Core\Field\FieldItemInterface.                                                                                      
         🪪  varTag.nativeType                                                                                                      
 ------ --------------------------------------------------------------------------------------------------------------------------- 

                                                                                                                        
 [ERROR] Found 4 errors                                                                                                 
                                                                                                                        
```

### To Reproduce
Steps to reproduce the behavior:

<!--- Edit and replace examples below. Add or remove as needed  -->

1. Clone fresh copy of az_quicsktart repo
2. Build locally with lando or ddev
3. Run `lando phpstan` or `ddev phpstan`
4. See errors

## Proposed resolution
Need to find the cause of the new failures to determine best resolution plan.

--- a/modules/custom/az_accordion/src/Plugin/Field/FieldFormatter/AZAccordionDefaultFormatter.php
+++ b/modules/custom/az_accordion/src/Plugin/Field/FieldFormatter/AZAccordionDefaultFormatter.php
@@ -9,6 +9,7 @@ use Drupal\Core\Field\FormatterBase;
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
 use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\az_accordion\Plugin\Field\FieldType\AZAccordionItem;
 use Drupal\paragraphs\ParagraphInterface;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
@@ -94,8 +95,8 @@ class AZAccordionDefaultFormatter extends FormatterBase implements ContainerFact
     $entity = $items->getEntity();
     $accordion_container_id = HTML::getUniqueId('accordion-' . $entity->id());
 
-    /** @var \Drupal\az_accordion\Plugin\Field\FieldType\AZAccordionItem $item */
     foreach ($items as $delta => $item) {
+      assert($item instanceof AZAccordionItem);
       // Format title.
       $title = $item->title ?? '';
 

--- a/modules/custom/az_card/src/Plugin/Field/FieldFormatter/AZCardDefaultFormatter.php
+++ b/modules/custom/az_card/src/Plugin/Field/FieldFormatter/AZCardDefaultFormatter.php
@@ -2,6 +2,7 @@
 
 namespace Drupal\az_card\Plugin\Field\FieldFormatter;
 
+use Drupal\az_card\Plugin\Field\FieldType\AZCardItem;
 use Drupal\Core\Field\Attribute\FieldFormatter;
 use Drupal\Core\Field\FieldItemListInterface;
 use Drupal\Core\Field\FormatterBase;
@@ -106,8 +107,8 @@ class AZCardDefaultFormatter extends FormatterBase implements ContainerFactoryPl
     $settings = $this->getSettings();
     $element = [];
 
-    /** @var \Drupal\az_card\Plugin\Field\FieldType\AZCardItem $item */
     foreach ($items as $delta => $item) {
+      assert($item instanceof AZCardItem);
 
       // Format title.
       $title = $item->title ?? '';

--- a/modules/custom/az_publication/src/Plugin/Field/FieldWidget/AZEntityRoleInlineFormComplex.php
+++ b/modules/custom/az_publication/src/Plugin/Field/FieldWidget/AZEntityRoleInlineFormComplex.php
@@ -2,6 +2,7 @@
 
 namespace Drupal\az_publication\Plugin\Field\FieldWidget;
 
+use Drupal\az_publication\Plugin\Field\FieldType\AZEntityRoleReferenceItem;
 use Drupal\Core\Entity\FieldableEntityInterface;
 use Drupal\Core\Field\FieldFilteredMarkup;
 use Drupal\Core\Field\FieldItemListInterface;
@@ -93,6 +94,7 @@ class AZEntityRoleInlineFormComplex extends InlineEntityFormComplex {
       // Store the $items entities in the widget state, for further
       // manipulation.
       foreach ($items as $delta => $item) {
+        assert($item instanceof AZEntityRoleReferenceItem);
         // Display the entity in the correct translation.
         $entity = $item->entity;
         $role = $item->role ?? 'author';

--- a/modules/custom/az_ranking/src/Plugin/Field/FieldFormatter/AZRankingDefaultFormatter.php
+++ b/modules/custom/az_ranking/src/Plugin/Field/FieldFormatter/AZRankingDefaultFormatter.php
@@ -2,6 +2,7 @@
 
 namespace Drupal\az_ranking\Plugin\Field\FieldFormatter;
 
+use Drupal\az_ranking\Plugin\Field\FieldType\AZRankingItem;
 use Drupal\Core\Field\Attribute\FieldFormatter;
 use Drupal\Core\Field\FieldItemListInterface;
 use Drupal\Core\Field\FormatterBase;
@@ -106,8 +107,8 @@ class AZRankingDefaultFormatter extends FormatterBase implements ContainerFactor
     $settings = $this->getSettings();
     $element = [];
 
-    /** @var \Drupal\az_ranking\Plugin\Field\FieldType\AZRankingItem $item */
     foreach ($items as $delta => $item) {
+      assert($item instanceof AZRankingItem);
 
       // Format title.
       $ranking_heading = $item->ranking_heading ?? '';

